impeller/entity/geometry/stroke_path_geometry.cc

```diff
@@ -328,10 +353,10 @@ StrokePathGeometry::CreateSolidStrokeVertices(
     }
 
     // Generate end cap or join.
-    if (!polyline.contours[contour_i].is_closed) {
+    if (!contour.is_closed) {
       auto cap_offset =
           Vector2(-contour.end_direction.y, contour.end_direction.x) *
-          stroke_width * 0.5;  // Clockwise normal
+          stroke_width * 0.5;
       cap_proc(vtx_builder, polyline.points[contour_end_point_i - 1],
                cap_offset, scale, false);
     } else {
```

Comment:
Now that we know when we're in a curve or not, we should also be able to get away with removing joins (not calling `join_proc` right below) for curves.